README.md

```diff
@@ -22,36 +36,235 @@ component written in native code. Using that component from DNX
 currently [requires a workaround][grpc-workaround]; on Windows please run `fix_dnx_grpc.bat`
 before running any gRPC code.
 
-Documentation
--------------
+## Specifying a Project ID
+
+Most `gcloud-dotnet` libraries require a project ID. If you don't remember yours (or haven't created a project yet), navigate to the [Google Developers Console][google-developers-console] to view your project ID (or create a new project and get the ID then). Once done, record the value and make sure to pass it as a parameter to the methods that require it.
+
+## Authentication
+Every API call needs to be authenticated. In order to successfully make a call, first ensure that the necessary Google Cloud APIs are enabled for your project and that you've downladed the right set of keys (if it applies to you) as explained in the [authentication document][cloud-common-authentication].
+
+Next, choose a method for authenticating API requests from within your project:
+
+1. When using `gcloud-dotnet` libraries from within Compute/App Engine, no additional authentication steps are necessary.
+2. When using `gcloud-dotnet` libraries elsewhere, you can do one of the following:
+    * Define the environment variable GOOGLE_APPLICATION_CREDENTIALS to be the location of the key.  For example:
+
+      ```
+      set GOOGLE_APPLICATION_CREDENTIALS=/path/to/my/key.json
+      ``` 
+    * If running locally for development/testing, you can authenticate using the [Google Cloud SDK][google-cloud-sdk].  Download the SDK if you haven't already, then login by running the following in the command line:
+
+      ```
+      gcloud auth login
+      ```
+
+## Google BigQuery
```

Comment:
Yes, we might do that later actually. If we start getting too many APIs also we'll re-adjust. And Jon has created an issue to be able to inject code example automatically into the ReadMes instead of manually writing. For now it makes the readme feel more alive by having the snippets there at the top level.